erpnext_shipping/erpnext_shipping/doctype/sendcloud/sendcloud.py

```diff
@@ -69,60 +90,159 @@ def get_available_services(self, delivery_address, parcels: list[dict]):
-			)
-			response_data = response.json()
-			if "failed_parcels" in response_data:
-				error = response_data["failed_parcels"][0]["errors"]
-				frappe.msgprint(
-					_("Error occurred while creating Shipment: {0}").format(error),
-					indicator="orange",
-					alert=True,
+		if service_info.get("multicollo"):
+			# Multicollo Logic: All packages are processed in a single API call
+			try:
+				response = requests.post(
+					SHIPMENTS_URL,
+					json=payload,
+					auth=(self.api_key, self.api_secret),
 				)
-			else:
-				shipment_id = ", ".join([str(x["id"]) for x in response_data["parcels"]])
-				awb_number = ", ".join([str(x["tracking_number"]) for x in response_data["parcels"]])
-				return {
+				response_data = response.json()
+				if "errors" in response_data and response_data["errors"]:
+					error_details = [
+						f"Code: {err.get('code', 'N/A')}, Detail: {err.get('detail', 'N/A')}"
+						for err in response_data["errors"]
+					]
+					error_message = "\n".join(error_details)
+					frappe.msgprint(
+						_("Error occurred while creating shipment: %(error_message)s")
+						% {"error_message": error_message},
+						indicator="red",
+						alert=True,
+					)
+					return None
+
+				parcels_data = response_data.get("data", {}).get("parcels", [])
+				if parcels_data:
+					shipment_ids = [str(parcel["id"]) for parcel in parcels_data]
+					tracking_numbers = [parcel.get("tracking_number") or "" for parcel in parcels_data]
+					tracking_urls = [parcel.get("tracking_url") or "" for parcel in parcels_data]
+					return {
+						"service_provider": "SendCloud",
+						"shipment_id": ", ".join(shipment_ids),
+						"carrier": self.get_carrier(service_info["carrier"], post_or_get="post"),
+						"carrier_service": service_info["service_name"],
+						"shipment_amount": service_info["total_price"],
+						"awb_number": ", ".join(tracking_numbers),
+						"tracking_url": ", ".join(tracking_urls),
+					}
+			except Exception:
+				show_error_alert("creating SendCloud Shipment (multicollo)")
+		else:
+			# Non-Multicollo Logic: A separate API call is made for each package
+			shipments_results = []
+			for parcel in parcels:
+				payload_single = payload.copy()
+				payload_single["parcels"] = [parcel]
+				try:
+					response = requests.post(
+						SHIPMENTS_ANNOUNCE_URL,
+						json=payload_single,
+						auth=(self.api_key, self.api_secret),
+					)
+					response_data = response.json()
+					if "errors" in response_data and response_data["errors"]:
+						error_details = [
+							f"Code: {err.get('code', 'N/A')}, Detail: {err.get('detail', 'N/A')}"
+							for err in response_data["errors"]
+						]
+						error_message = "\n".join(error_details)
+						frappe.msgprint(
+							_(
+								"Error occurred while creating shipment for parcel %(order_number)s:\n%(error_message)s"
+							)
+							% {
+								"order_number": parcel.get("order_number"),
+								"error_message": error_message,
+							},
```

Comment:
```suggestion
							_(
								"Error occurred while creating shipment for parcel {0}:\n{1}"
							).format(
								parcel.get("order_number"),
								error_message
							),
```

We usually use `.format()` for string formatting. This is a common practice across frappe apps.